src/node.rs

```diff
@@ -1,9 +1,112 @@
 use std::fmt;
 use std::ops::Bound;
 use std::ops::Bound::*;
+#[cfg(any(feature="serde", test))]
+use serde::{Serialize, Deserialize};
 
 pub(crate) type Range<K> = (Bound<K>, Bound<K>);
 
+#[cfg(test)]
```

Comment:
Could you please move the tests to the bottom of this file. I prefer to have the implementation before tests.